src/utils/logging.ts

```diff
@@ -22,9 +22,23 @@ const paths = XDGAppPaths({ name: 'rancher-desktop' });
 const logDir = path.join(paths.runtime() || paths.state(), 'logs');
 
 interface Log {
+  /**
+   * Log a message to the log file.
+   */
   (message: string): Promise<void>;
+  /**
+   * The path to the log file.
+   */
   path: string;
+  /**
+   * A stream to write to the log file.
+   */
   stream: stream.Writable;
+  /**
```

Comment:
Where is this guarantee met?

Reply:
[Here] — writing to the file requires that we have a `fd` to write to.  Since we resolve the promise on successful write, at that point we'll always have a `fd`.  This is effectively moving the [old code].

[Here]: https://github.com/rancher-sandbox/rd/pull/261/files/85829da234a73404880606a944e4276f7ef0fe33#diff-0e07a00003db43dce30c641a3a8584a95408101eb80a17dc8f0e6aaf029b77c2R73
[old code]: https://github.com/rancher-sandbox/rd/pull/261/files/85829da234a73404880606a944e4276f7ef0fe33#diff-2ffd77af22645fa7b6fd0cc0b6d6b28c1958dda2aad92457580552406f1f7386L376